yellowbrick/cluster/silhouette.py

```diff
@@ -97,13 +104,15 @@ class SilhouetteVisualizer(ClusteringScoreVisualizer):
     >>> model.poof()
     """
 
-    def __init__(self, model, ax=None, **kwargs):
+    def __init__(self, model, ax=None, colormap='set1', colors=None, **kwargs):
         super(SilhouetteVisualizer, self).__init__(model, ax=ax, **kwargs)
 
         # Visual Properties
-        # TODO: Fix the color handling
-        self.colormap = kwargs.get('colormap', 'set1')
-        self.color = kwargs.get('color', None)
+        # Use colors if it is given, otherwise attempt to find a colormap in
+        # styles.PALETTES. If not found, default to None. The colormap may
+        # yet still be found in resolve_colors
+        self.colormap = colormap
```

Comment:
change this to self.colormap = kwargs.get('colormap', 'set1')

What this does is sets the colormap to the user defined colormap, otherwise sets the colormap to set1 if the user does not provide a colormap.